### PR TITLE
8343150: Change URLClassLoader.getPermissions to return empty PermissionCollection

### PR DESCRIPTION
**Pull request by jaikiran**
Can I please get a review of this change which removes the `getPermissions()` method from the `URLClassLoader`? This addresses https://bugs.openjdk.org/browse/JDK-8343150.

With the SecurityManager implementation removed through JEP 486, this method of the `URLClassLoader` is no longer relevant.

No new tests have been added. tier1 and tier2 testing is currently in progress.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8344354](https://bugs.openjdk.org/browse/JDK-8344354) to be approved

### Issues
 * [JDK-8343150](https://bugs.openjdk.org/browse/JDK-8343150): Change URLClassLoader.getPermissions to return empty PermissionCollection (**Enhancement** - P4)
 * [JDK-8344354](https://bugs.openjdk.org/browse/JDK-8344354): Change URLClassLoader.getPermissions to return empty PermissionCollection (**CSR**)

### Reviewers
 * [Michael McMahon](https://openjdk.org/census#michaelm) (@Michael-Mc-Mahon - **Reviewer**) Review applies to [e5d8ee3b](https://git.openjdk.org/jdk/pull/22179/files/e5d8ee3b1c477279f6bba976c381d16a6fecf774)
 * [Daniel Fuchs](https://openjdk.org/census#dfuchs) (@dfuch - **Reviewer**)
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22179/head:pull/22179` \
`$ git checkout pull/22179`

Update a local copy of the PR: \
`$ git checkout pull/22179` \
`$ git pull https://git.openjdk.org/jdk.git pull/22179/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22179`

View PR using the GUI difftool: \
`$ git pr show -t 22179`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22179.diff">https://git.openjdk.org/jdk/pull/22179.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22179#issuecomment-2480919947)
</details>

### CODE REVIEW

**Comment by jaikiran** [comment]
tier1, tier2 and tier3 testing completed without related failures with this change.

**Review by Michael-Mc-Mahon — APPROVED** [review]
Looks fine.

File: src/java.base/share/classes/java/security/SecureClassLoader.java [hunk 1]
```diff
@@ -165,16 +165,8 @@ protected final Class<?> defineClass(String name, java.nio.ByteBuffer b,
...
-     * Returns the permissions for the given CodeSource object.
-     * <p>
-     * This method is invoked by the defineClass method which takes
-     * a CodeSource as an argument when it is constructing the
-     * ProtectionDomain for the class being defined.
-     *
-     * @param codesource the codesource.
-     *
-     * @return the permissions for the codesource.
-     *
```
Comment (Michael-Mc-Mahon) [inline src/java.base/share/classes/java/security/SecureClassLoader.java:177]: This change will have to go in the CSR as well presumably.

Comment (jaikiran) [reply src/java.base/share/classes/java/security/SecureClassLoader.java:177]: Hello Michael, right - I have been talking to Alan and Sean about this change and there has been some inputs from Joe in the CSR too. This latest change to the SecureClassLoader was one proposal that's being considered. I suspect we will need at least one more day to come to a consensus on how we want to proceed. But yes, if we do decide to make this change, I'll update the CSR text.

File: src/java.base/share/classes/java/net/URLClassLoader.java [hunk 2]
```diff
@@ -573,84 +569,15 @@ public boolean hasMoreElements() {
...
-     * that directory.
-     * <p>
-     * If the protocol is not "file", then permission
-     * to connect to and accept connections from the URL's host is returned.
-     * @param codesource the codesource
-     * @throws    NullPointerException if {@code codesource} is {@code null}.
-     * @return the permissions for the codesource
+     * {@return an {@linkplain PermissionCollection empty Permission collection}}
+     *
+     * @param codesource the CodeSource
```
Comment (dfuch) [inline src/java.base/share/classes/java/net/URLClassLoader.java:574]:
```suggestion
     * @param codesource the {@code CodeSource}
```

Comment (jaikiran) [reply src/java.base/share/classes/java/net/URLClassLoader.java]: Done.

**Comment by jaikiran** [comment]
After further discussions between Alan, Sean and me, we decided to retain the `getPermissions()` method on the `URLClassLoader` and instead respecify it to return an empty `PermissionCollection`. The CSR and the PR has been updated accordingly.

**Review by dfuch — APPROVED** [review]
LGTM

**Review by AlanBateman — APPROVED** [review]

**Comment by jaikiran** [comment]
The CSR has been approved. I'll go ahead and integrate this now. Thank you all for the reviews and the help in deciding what changes should be done here.

**Pull request closed without merging**